cannot use innermost value of `map[string][1][2]map[string]int`

**V version:**
master
 
**OS:**
win10 

**What did you do?**
```V
fn main() {
	mut m := map[string][1][2]map[string]int
	m['foo'] = [[{'bar': 1}, {'baz': 3}]!]!
	println(m['foo'][0][0]['bar'])
        println(m['foo'][0][0]['bar'] == 1)
}
```

**What did you expect to see?**
1
true
 
**What did you see instead?**
```
C:\Users\Roy\AppData\Local\Temp\v\sample.14733716196300629677.tmp.c: In function 'main__main':
C:\Users\Roy\AppData\Local\Temp\v\sample.14733716196300629677.tmp.c:8521:227: error: expected expression before ')' token
  println(int_str(map_set_1(&(*(array_fixed_array_fixed_map_string_int_2_1*)map_get_and_set_1(&m, &(string[]){_SLIT("foo")}, &(array_fixed_array_fixed_map_string_int_2_1[]){ {0} }))[0][0], &(string[]){_SLIT("bar")}, &(int[]) { ));
                                                                                                                                                                                                                                   ^
C:\Users\Roy\AppData\Local\Temp\v\sample.14733716196300629677.tmp.c:8521:229: error: expected ')' before ';' token
  println(int_str(map_set_1(&(*(array_fixed_array_fixed_map_string_int_2_1*)map_get_and_set_1(&m, &(string[]){_SLIT("foo")}, &(array_fixed_array_fixed_map_string_int_2_1[]){ {0} }))[0][0], &(string[]){_SLIT("bar")}, &(int[]) { ));
                                                                                                                                                                                                                                     ^
                                                                                                                                                                                                                                     )
C:\Users\Roy\AppData\Local\Temp\v\sample.14733716196300629677.tmp.c:8522:283: error: expected ';' before '}' token
  println((*(int*)map_get_1(ADDR(map, (*(array_fixed_array_fixed_map_string_int_2_1*)map_get_1(ADDR(map, m), &(string[]){_SLIT("foo")}, &(array_fixed_array_fixed_map_string_int_2_1[]){ {0} }))[0][0]), &(string[]){_SLIT("bar")}, &(int[]){ 0 })) == 1 ? _SLIT("true") : _SLIT("false"));
                                                                                                                                                                                                                                                                                           ^
                                                                                                                                                                                                                                                                                           ;
 }
 ~
builder error:
==================
C error. This should never happen.
```

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -2005,6 +2005,7 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 					}
 				}
 			}
+			g.is_assign_lhs = false
 		} else {
 			is_inside_ternary := g.inside_ternary != 0
 			cur_line := if is_inside_ternary && is_decl {

--- a/vlib/v/tests/map_complex_fixed_array_test.v
+++ b/vlib/v/tests/map_complex_fixed_array_test.v
@@ -13,7 +13,7 @@ fn test_innermost_value_of_map_fixed_array() {
 	mut m := map[string][1][2]map[string]int
 	m['foo'] = [[{'bar': 1}, {'baz': 3}]!]!
 	println(m['foo'][0][0]['bar'])
-    println(m['foo'][0][0]['bar'] == 1)
+	println(m['foo'][0][0]['bar'] == 1)
 	assert m['foo'][0][0]['bar'] == 1
 	assert '${m['foo'][0][0]['bar']}' == '1'
 }

--- a/vlib/v/tests/map_complex_fixed_array_test.v
+++ b/vlib/v/tests/map_complex_fixed_array_test.v
@@ -8,3 +8,12 @@ fn test_complex_map_fixed_array() {
 	println(m)
 	assert '$m' == "{'foo': [[{'bar': 1}, {'baz': 3}]]}"
 }
+
+fn test_innermost_value_of_map_fixed_array() {
+	mut m := map[string][1][2]map[string]int
+	m['foo'] = [[{'bar': 1}, {'baz': 3}]!]!
+	println(m['foo'][0][0]['bar'])
+    println(m['foo'][0][0]['bar'] == 1)
+	assert m['foo'][0][0]['bar'] == 1
+	assert '${m['foo'][0][0]['bar']}' == '1'
+}

